File: src/runtime/ruby/std/math_native.rb
# Generated std/math.rb delegates host bindings through this native seam.
# source: src/runtime/cs/std/math_native.cs (reference)

module MathNative
  def self.pi = Math::PI
  def self.e = Math::E
  def self.sqrt(x) = Math.sqrt(x.to_f)
  def self.sin(x) = Math.sin(x.to_f)
  def self.cos(x) = Math.cos(x.to_f)
  def self.tan(x) = Math.tan(x.to_f)
  def self.exp(x) = Math.exp(x.to_f)
  def self.log(x) = Math.log(x.to_f)
  def self.log10(x) = Math.log10(x.to_f)
  def self.fabs(x) = x.to_f.abs
  def self.floor(x) = x.to_f.floor.to_f
  def self.ceil(x) = x.to_f.ceil.to_f
  def self.pow(x, y) = x.to_f ** y.to_f
end

File: src/runtime/lua/std/math_native.lua
-- Generated std/math.lua delegates host bindings through this native seam.
-- source: src/runtime/cs/std/math_native.cs (reference)

local math_native = {}

math_native.pi = math.pi
math_native.e = 2.718281828459045

function math_native.sqrt(x) return math.sqrt(tonumber(x) or 0) end
function math_native.sin(x) return math.sin(tonumber(x) or 0) end
function math_native.cos(x) return math.cos(tonumber(x) or 0) end
function math_native.tan(x) return math.tan(tonumber(x) or 0) end
function math_native.exp(x) return math.exp(tonumber(x) or 0) end
function math_native.log(x) return math.log(tonumber(x) or 0) end
function math_native.log10(x) return math.log(tonumber(x) or 0, 10) end
function math_native.fabs(x) return math.abs(tonumber(x) or 0) end
function math_native.floor(x) return math.floor(tonumber(x) or 0) end
function math_native.ceil(x) return math.ceil(tonumber(x) or 0) end
function math_native.pow(x, y) return (tonumber(x) or 0) ^ (tonumber(y) or 0) end

return math_native
